build.gradle

```diff
@@ -27,131 +24,24 @@ mamsdkProperties.load(new FileInputStream(mamsdkPropertiesFile))
-    }
-    println 'mamSdkVersion:' + mamSdkVersion
-
     appPackageName = mamsdkProperties['appPackageName']
-    if (appPackageName.isEmpty()) {
-        throw new GradleException('ERROR: appPackageName property value is missing in mamsdk.properties!!!')
-    }
-    println 'appPackageName:' + appPackageName
-
     keyStorePath = keystoreProperties['keyStorePath']
-    if (keyStorePath.isEmpty()) {
-        throw new GradleException('ERROR: keyStorePath property value is missing in keystore.properties!!!')
-    }
-    println 'keyStorePath:' + keyStorePath
-
     keystorePassword = keystoreProperties['keystorePassword']
-    if (keystorePassword.isEmpty()) {
-        throw new GradleException('ERROR: keystorePassword property value is missing in keystore.properties!!!')
-    }
     keyAlias = keystoreProperties['keyAlias']
-    if (keyAlias.isEmpty()) {
-        throw new GradleException('ERROR: keyAlias property value is missing in keystore.properties!!!')
-    }
     keyPassword = keystoreProperties['keyPassword']
-    if (keyPassword.isEmpty()) {
-        throw new GradleException('ERROR: keyPassword property value is missing in keystore.properties!!!')
-    }
-}
-
-configurations.all {
-    // Check for updates every build
-    resolutionStrategy.cacheChangingModulesFor 0, 'seconds'
 }
 
-android {
-    compileSdkVersion 30
-
-    defaultConfig {
-        applicationId "${project.ext.appPackageName}"
-        minSdkVersion 24
-        targetSdkVersion 30
-        versionCode 1
-        versionName "1.0"
-        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
-    }
-
-    signingConfigs {
-        release {
-            storeFile file(project.ext.keyStorePath)
-            storePassword "$project.ext.keystorePassword"
-            keyAlias "$project.ext.keyAlias"
-            keyPassword "$project.ext.keyPassword"
-        }
-
-        debug {
-            storeFile file(project.ext.keyStorePath)
-            storePassword "$project.ext.keystorePassword"
-            keyAlias "$project.ext.keyAlias"
-            keyPassword "$project.ext.keyPassword"
-        }
-    }
-
-    compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
-    }
-
+allprojects {
     repositories {
-        maven { url "https://raw.githubusercontent.com/citrix/citrix-mam-sdks/main/maven" }
+        maven { url "$rootProject.ext.mamSdkLibraryMaven" }
         google()
-        jcenter()
-    }
-
-    buildTypes {
-        release {
-            minifyEnabled false
-            proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
-            signingConfig signingConfigs.release
-        }
-        debug {
-            debuggable true
-            signingConfig signingConfigs.debug
-        }
+        mavenCentral()
+        jcenter() // Warning: this repository is going to shut down soon
```

Comment:
Might as well remove it?

Reply:
Release 20.10.5.1 will work without jcenter(). But the next release has some dependency with jcenter(), so we will need to consider removing jcenter() dependency in future.